exonum/src/events/network.rs

```diff
@@ -489,28 +563,7 @@ impl NetworkHandler {
     }
 
     fn can_create_connections(&self) -> bool {
-        self.pool.len() <= self.network_config.max_outgoing_connections
-    }
-
-    fn disconnect_with_peer(
-        &self,
-        peer: PublicKey,
-    ) -> impl Future<Item = (), Error = failure::Error> {
-        if self.pool.remove(&peer).is_some() {
-            to_box(
-                self.network_tx
-                    .clone()
-                    .send(NetworkEvent::PeerDisconnected(peer))
-                    .map_err(|_| format_err!("can't send disconnect"))
-                    .map(drop),
-            )
-        } else {
-            Box::new(err(format_err!(
-                "Attempt to connect to disconnect with peer with key {:?} which \
-                 is not in the connection pool",
-                peer
-            )))
-        }
+        self.pool.len() < self.network_config.max_outgoing_connections
```

Comment:
I remember using `<` vs `<=` here has some history. Is this change intentional?

Reply:
Yes, here it should be this way.